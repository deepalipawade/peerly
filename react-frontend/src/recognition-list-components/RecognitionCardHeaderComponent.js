import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import { Row, Col } from "core-components/grid/GridComponent";
import { Form } from "core-components/form/FormComponent";
import { Button } from "core-components/button/ButtonComponent";
import ImageComponent from "core-components/image/ImageComponent";

const onclickMenuButton = () => {
  //todo function for menu
};

const Img = styled.div`
  width: 12vh;
`;

const RecognitionCardHeaderComponent = ({
  given_at,
  given_for,
  core_value,
}) => {
  return (
    <Row>
      <Col>
        <Row className="d-flex justify-content-end">
          <Button
            className="bg-white font-weight-bold text-dark btn-outline-light grey"
            onClick={onclickMenuButton}
          >
            ...
          </Button>
        </Row>
        <Row className="d-flex flex-column-left">
          <Img>
            <ImageComponent
              src="https://i.picsum.photos/id/2/200/200.jpg"
              roundedCircle={true}
              alt="Profile"
            />
          </Img>
          <Row className="d-flex flex-column text-start ml-4">
            <Form.Label className="font-weight-bold ">{given_for}</Form.Label>
            <Row className=" d-sm-block d-xs-block d-md-none">
              <Row className="d-flex flex-column ml-2">
                <Form.Label>got a high five for</Form.Label>
                <Form.Label className="font-weight-bold ml-1">
                  {core_value}
                </Form.Label>
              </Row>
            </Row>
            <Row className="d-none d-md-block d-{sm,xs}-none">
              <Row className="d-flex flex-row ml-2">
                <Form.Label>got a high five for</Form.Label>
                <Form.Label className="font-weight-bold ml-1">
                  {core_value}
                </Form.Label>
              </Row>
            </Row>
            <Form.Label className="text-muted">{given_at}</Form.Label>
          </Row>
        </Row>
      </Col>
    </Row>
  );
};
RecognitionCardHeaderComponent.propTypes = {
  given_for: PropTypes.string.isRequired,
  given_at: PropTypes.string.isRequired,
  core_value: PropTypes.string.isRequired,
};

export default React.memo(RecognitionCardHeaderComponent);
