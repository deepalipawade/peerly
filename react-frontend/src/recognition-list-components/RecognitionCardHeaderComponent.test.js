import React from "react";
import { render } from "@testing-library/react";

import RecognitionCardHeaderComponent from "recognition-list-components/RecognitionCardHeaderComponent";

describe("Recognition TopBar Component test", () => {
  test("should equal to recognition name", () => {
    const { getByText } = render(
      <RecognitionCardHeaderComponent
        given_at="today 3:00 AM"
        given_for="avinash"
        core_value="positive"
      />
    );
    expect(getByText("avinash")).toHaveTextContent("avinash");
  });

  test("should equal to recognition time", () => {
    const { getByText } = render(
      <RecognitionCardHeaderComponent
        given_at="today 3:00 AM"
        given_for="avinash"
        core_value="positive"
      />
    );
    expect(getByText("today 3:00 AM")).toHaveTextContent("today 3:00 AM");
  });

  test("render image component", () => {
    const { getByAltText } = render(
      <RecognitionCardHeaderComponent
        given_at="today 3:00 AM"
        given_for="avinash"
        core_value="positive"
      />
    );
    expect(getByAltText("Profile")).toBeInTheDocument();
  });
});
