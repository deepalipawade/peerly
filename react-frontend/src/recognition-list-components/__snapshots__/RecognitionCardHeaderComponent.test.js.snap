// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recognition card header Component test should equal snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="d-flex justify-content-end"
    >
      <button
        class="bg-white font-weight-bold text-dark btn-outline-light btn btn-primary"
        type="button"
      >
        ...
      </button>
    </div>
    <div
      class="d-flex flex-row"
    >
      <div
        class="sc-AxjAm fxHfgf"
      >
        <img
          alt="Profile"
          class="img-fluid rounded-circle"
          src="https://i.picsum.photos/id/2/200/200.jpg"
        />
      </div>
      <div
        class="text-start ml-2"
      >
        <label
          class="font-weight-bold  form-label"
        >
          Roy
        </label>
        <div>
          <label
            class="form-label"
          >
             got a high five for
          </label>
          <label
            class="ml-1 font-weight-bold form-label"
          >
            positive
          </label>
        </div>
        <label
          class="text-muted form-label"
        >
          Today 10:30 AM
        </label>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
